### ISSUE TITLE
MakeBucket shouldn't create buckets on unformatted disks

### ISSUE THREAD

**Issue by krisis** [minio/minio#6570]
`mc mb ALIAS/bucket` creates directories for buckets on all disks including unformatted fisks in an erasure code setup.

## Expected Behavior

All bucket and object operations must be performed only on formatted disks in an erasure code setup.
## Current Behavior
MakeBucket creates directories even on unformatted disks.

## Possible Solution

Check if the disk where I/O is performed is formatted before including it in the set of available disks.

## Steps to Reproduce (for bugs)

```
0. minio server /tmp/{1..4}
1. mc mb ALIAS/bucket1
2. rm -rf /tmp/1 --> same as losing a disk
3. mkdir /tmp/1 ---> same as replacing the disk
4. mc mb ALIAS/bucket2
5. tree /tmp/1 --> you will see a directory for bucket1 created with no format.json present.
```
## Regression

Don't know yet

## Your Environment

* Version used (`minio version`):
* Environment name and version (e.g. nginx 1.9.1):
* Server type and version:
* Operating System and version (`uname -a`):
* Link to your project:

**Comment by harshavardhana** [comment]
The main issue here is that is hot swapping of disks underneath an active server supported or not, we actually do not work when that happens at least from an implementation point of view.

Usually, the steps taken to replace disks are as below in a production environment. Any server i.e having its disks replaced goes into maintenance mode, usually to avoid i/o an admin takes the server down. Replaces the disk and starts Minio again. All good everything works no i/o goes to the server.

Now on really dense servers say with > 24 drives this can be a problem, a downtime for an application for a single disk replacement is not expected and considered a limitation. 

In the above reproduction what has really happened is an equivalent of hot swapping drives, if we do need to support this scenario then we have to fix this issue through a fundamental change in how we consider a drive to be available for use. If not what we have today is correct. 

What we really need is a mechanism on Minio without taking down the server down to tell server if the disk is going to be in maintenance mode, either repaired (xfs_repair), or being replaced etc. Such that when the disk is replaced it doesn't really become automatically available for use as is observed in this issue. 

Blocked on @abperiasamy for further discussion.

**Comment by harshavardhana** [comment]
This is fixed in master #7229
